File: apps/backend/src/main/java/io/secureai/backend/domain/dashboard/dto/DashboardResponse.java
package io.secureai.backend.domain.dashboard.dto;

import java.util.List;
import java.util.Map;

/**
 * 대시보드 집계 응답 DTO.
 * 5개의 차트 데이터를 단일 엔드포인트로 반환한다.
 */
public record DashboardResponse(
        int securityScore,
        SeverityCounts severityCounts,
        List<TrendPoint> trend,
        List<FileHeatPoint> fileHeatmap,
        Map<String, Boolean> owaspCoverage
) {

    /** 심각도별 취약점 건수 */
    public record SeverityCounts(
            long critical,
            long high,
            long medium,
            long low,
            long info
    ) {}

    /** 일별 취약점 발생 추이 */
    public record TrendPoint(String date, long count) {}

    /** 파일별 취약점 집중도 */
    public record FileHeatPoint(String filePath, long count) {}
}

File: apps/android/app/src/main/java/io/secureai/android/data/remote/model/DashboardResponse.kt
package io.secureai.android.data.remote.model

data class DashboardResponse(
    val securityScore: Int,
    val severityCounts: SeverityCounts,
    val trend: List<TrendPoint>,
    val fileHeatmap: List<FileHeatPoint>,
    val owaspCoverage: Map<String, Boolean>
)

data class SeverityCounts(
    val CRITICAL: Int,
    val HIGH: Int,
    val MEDIUM: Int,
    val LOW: Int,
    val INFO: Int
)

data class TrendPoint(
    val date: String,
    val count: Int
)

data class FileHeatPoint(
    val filePath: String,
    val count: Int
)
